hummingbot/connector/exchange/xrpl/xrpl_exchange.py

```diff
@@ -0,0 +1,1261 @@
+            return None
+
+        if tx.get("TransactionType") != "OfferCreate":
+            return None
+
+        offer_changes = get_order_book_changes(meta)
+        balance_changes = get_balance_changes(meta)
+
+        # Filter out change that is not from this account
+        offer_changes = [x for x in offer_changes if
+                         x.get("maker_account") == self._auth.get_account()]
+        balance_changes = [x for x in balance_changes if
+                           x.get("account") == self._auth.get_account()]
+
+        tx_sequence = tx.get("Sequence")
+
+        if int(tx_sequence) == int(sequence):
+            # check status of the transaction
+            tx_status = meta.get("TransactionResult")
+            if tx_status != "tesSUCCESS":
+                self.logger().error(
+                    f"Order {order.client_order_id} ({order.exchange_order_id}) failed: {tx_status}")
+                return None
+
+            # check if there is no offer changes, if none, this order has been filled
+            if len(offer_changes) == 0:
+                # check if there is any balance changes
+                if len(balance_changes) == 0:
+                    self.logger().error(
+                        f"Order {order.client_order_id} ({order.exchange_order_id}) has no balance changes")
+                    return None
+
+                for balance_change in balance_changes:
+                    changes = balance_change.get("balances", [])
+                    base_change = get_token_from_changes(changes, token=base_currency.currency)
+                    quote_change = get_token_from_changes(changes, token=quote_currency.currency)
+
+                    if order.trade_type is TradeType.BUY:
+                        fee_token = fee_rules.get("quote_token")
+                        fee_rate = fee_rules.get("quote_transfer_rate")
+                    else:
+                        fee_token = fee_rules.get("base_token")
+                        fee_rate = fee_rules.get("base_transfer_rate")
+
+                    fee = TradeFeeBase.new_spot_fee(
+                        fee_schema=self.trade_fee_schema(),
+                        trade_type=order.trade_type,
+                        percent_token=fee_token.upper(),
+                        percent=Decimal(fee_rate),
+                    )
+                    trade_update = TradeUpdate(
+                        trade_id=tx.get("hash"),
+                        client_order_id=order.client_order_id,
+                        exchange_order_id=order.exchange_order_id,
+                        trading_pair=order.trading_pair,
+                        fee=fee,
+                        fill_base_amount=abs(Decimal(base_change.get("value"))),
+                        fill_quote_amount=abs(Decimal(quote_change.get("value"))),
+                        fill_price=abs(Decimal(base_change.get("value"))) / abs(
```

Comment:
This should be quote_change / base_change instead? Looking at the logs, the trade fill prices are way off, and that's probably the cause of the inconsistent PNL